satisfactory/chat.py

```diff
@@ -141,4 +163,36 @@ async def send_random_quote(self, context: CallbackContext) -> None:
     async def check_videos(self, context: CallbackContext) -> None:
         videos = self.youtube_client.get_new_videos()
         for video in videos:
-            await context.bot.send_message(chat_id=self.chat_id, text=video, parse_mode=ParseMode.HTML, disable_notification=True)
+            await context.bot.send_message(chat_id=self.chat_id, text=video, parse_mode=ParseMode.HTML,
+                                           disable_notification=True)
+
+    @Summary("satisfactory_bot_check_server_status", "Time spent running check_server_status").time()
+    async def check_server_status(self, context: CallbackContext) -> None:
+        if self.gameserver_address is None or self.gameserver_token is None:
+            return
+
+        try:
+            from pyfactorybridge import API
+            satisfactory_api_client = API(address=self.gameserver_address, token=self.gameserver_token)
+
+            server_stats = satisfactory_api_client.query_server_state()
+            server_stats = server_stats.get(KEY_SERVER_GAME_STATE, {})
+            new_num_connected_players = int(server_stats.get(KEY_SERVER_STATS_NUM_CONNECTED_PLAYERS, 0))
+            old_num_connected_players = self.last_server_stats.get(KEY_SERVER_STATS_NUM_CONNECTED_PLAYERS, 0) if self.last_server_stats else 0
+
+            text = None
+            if new_num_connected_players > 0 and new_num_connected_players != old_num_connected_players:
+                text = f"New active players: {new_num_connected_players}"
+            elif new_num_connected_players == 0 and new_num_connected_players != old_num_connected_players:
+                text = f"Server is empty."
+
+            if text is not None:
+                await context.bot.send_message(
+                    chat_id=self.chat_id, text=text,
+                    parse_mode=ParseMode.HTML, disable_notification=new_num_connected_players > 0
```

Comment:
Does this mean we only sent with notifications enabled if the server is empty? Wouldn't it be more useful to enable notifications when the server is not empty?

Reply:
yes, that was the intention. I guess I didn't see the "disable_" prefix :facepalm: